### PR TITLE
Add checker for snapcraft packages

### PR DESCRIPTION
**Pull request by FakeShemp**
<s>This currently doesn't work, but I want some feedback before fixing it.

Snap uses sha3-384 for hashes. Afaik, this is not supported in a Flatpak. How should this be handled? Add support for sha3-384 in Flatpak or try to get another hash some other way?</s>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,6 +193,20 @@ for [JetBrains](https://www.jetbrains.com/) products:
 }
 ```
 
+#### SnapCraft checker
```
Comment (barthalion) [inline README.md]: Snapcraft, not SnapCraft.

File: README.md [hunk 2]
```diff
@@ -193,6 +193,20 @@ for [JetBrains](https://www.jetbrains.com/) products:
...
+
+Special checker that will check for available updates
+for [SnapCraft](https://snapcraft.io/) packages:
+
+```json
+"x-checker-data": {
+    "type": "snapcraft",
+    "name": "PACKAGE-NAME",
+    "release": "stable, beta, or any other tag the project uses"
```
Comment (barthalion) [inline README.md]: This should be `channel`, at least according to snapcraft's dictionary.

**Comment by barthalion** [comment]
Seems flatpak-builder supports sha384. Is it the same thing that sha3-384?

**Comment by FakeShemp** [comment]
@barthalion Unfortunately not. That would be sha2-384.

**Comment by barthalion** [comment]
> @barthalion Unfortunately not. That would be sha2-384.

I don't think there's a good way to implement it in flatpak-builder, it's not supported by glib either if I see correctly. The checker should check if the checksum matches and rewrite it to sha256 if yes.

**Comment by FakeShemp** [comment]
Is this stuck at something special I can fix?

File: tests/test_snapcraftchecker.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import os
+import unittest
+
+from src.checker import ManifestChecker
+from src.lib.utils import init_logging
+
+TEST_MANIFEST = os.path.join(os.path.dirname(__file__), "com.nordpass.NordPass.yaml")
+
+
+class TestJetBrainsChecker(unittest.TestCase):
```
Comment (barthalion) [inline tests/test_snapcraftchecker.py]: Should be TestSnapctaftChecker.

**Comment by barthalion** [comment]
Sorry, I forgot about it. Just one minor thing.

**Pull request merged**